lib/Compilation.js

```diff
@@ -2799,4 +2864,72 @@ class Compilation {
 	}
 }
 
+/**
+ * Add additional assets to the compilation.
+ */
+Compilation.PROCESS_ASSETS_STAGE_ADDITIONAL = -2000;
+
+/**
+ * Basic preprocessing of assets.
+ */
+Compilation.PROCESS_ASSETS_STAGE_PRE_PROCESS = -1000;
+
+/**
+ * Derive new assets from existing assets.
+ * Existing assets should not be treated as complete.
+ */
+Compilation.PROCESS_ASSETS_STAGE_DERIVED = -200;
+
+/**
+ * Add additional sections to existing assets, like a banner or initialization code.
+ */
+Compilation.PROCESS_ASSETS_STAGE_ADDITIONS = -100;
```

Comment:
maybe `0` will be better, this way plugins that already use this hook for additions won't change

Reply:
I placed 0 just before OPTIMIZE, as existing plugins could use `optimizeAssets` and they will have 0 as stage.